### DISCUSSION TITLE
Is there a way to partially migrate to Teal?

### DISCUSSION THREAD

**Discussion by kristijanhusak** [teal-language/tl#594] [Q&A]
Hi,

I'm thinking about migrating my project (https://github.com/nvim-orgmode/orgmode) to Teal, but since the project is fairly big at this point, I would like to do it partially and slowly add more and more types.

Is there a way to set up compile step to work similarly to how Typescript works with `allowJs: true` ?  
Basically copy all `.lua` files to `teal` folder, and have a compiler compile any `tl` files that are in `teal` folder to `lua` folder as `.lua` files,  and for `lua` files that are in `teal` folder just to copy them over to the build path without actually doing any compiling.
I can probably achieve this with some custom script that would know to compile teal + copy `.lua` files, but I want to check first if it's possible to configure compiler to do that for me.

**Comment by lenscas** [comment]
I... think teal already does something like this?

However, it requires a `.d.tl` file next to the `.lua` file if a teal script wants to load a `.lua` file. So, probably best to start at the bottom and work your way up.

**Reply by kristijanhusak** [reply]
Could you elaborate a bit more on this? I can't seem to find anything in the docs that explains this, might be missing something.

**Reply by lenscas** [reply]
https://github.com/teal-language/tl/blob/master/docs/tutorial.md#using-tl-with-lua

goes a bit over it. To me it sounds like it should pretty much work out of the box.

**Reply by kristijanhusak** [reply]
It seems that this covers only type-checking. I was wondering if `tl build` knows to type-check+build `.tl` files into the destination directory, and also "build" `.lua` files into the destination directory. For example, when I run `tl build`, I would like to get something like this:

```lua
-- source dir
teal/
  file1.tl
  file2.lua

-- destination dir
lua/
  file1.lua <- This was compiled from `file1.tl` file
  file2.lua <- This was only copied from `teal` folder
```

**Reply by bjornbm** [reply]
Just a note that I understand that `tl build` is deprecated in favor of the [cyan](https://github.com/teal-language/cyan) build tool. See https://github.com/teal-language/tl/issues/579#issuecomment-1306091457

**Reply by lenscas** [reply]
yep, `tl build` is getting deprecated. Also, going by https://github.com/teal-language/cyan/issues/26 it sounds like `cyan` already supports this (else there isn't a way for lua source files to get overwritten?)
